### PR TITLE
Functional Api tests

### PR DESCRIPTION
**Pull request by hckenmiller**
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/235)

### CODE REVIEW

File: Fabric.Identity.API/scripts/CatalystDosIdentity.psm1 [hunk 1]
```diff
@@ -746,22 +745,21 @@ function Edit-ApiRegistration {
...
         $headers.Add("Authorization", "Bearer $accessToken")
     }
     try{
-        Invoke-RestMethod -Method Put -uri "$url" -body $body -ContentType "application/json" -Headers $headers | out-null
+        Invoke-RestMethod -Method Put -uri $url -body $body -ContentType "application/json" -Headers $headers | out-null
 
         # Reset api secret
-        $apiResponse = Reset-ApiPassword -identityUrl $url -apiName $apiName -accessToken $accessToken
-        return $apiResponse.apiSecret
+        $apiResponse = Reset-ApiPassword -identityUrl $($identityUrl.OriginalString) -apiName $apiName -accessToken $accessToken
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: May need to TrimEnd here as well

File: Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1 [hunk 2]
```diff
@@ -106,15 +105,150 @@ Describe 'Identity Cli Functional Tests' {
...
+
+                # Test an api that is registered
+                $testApi = Test-IsApiRegistered -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $testApi | Should -Be $true
+
+                # Get an api that is registered
+                $newApi = Get-ApiRegistration -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $newApi | Should -Not -Be $null
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1]: Should this also check to see if it returns some values that are expected?

File: Fabric.Identity.API/scripts/CatalystDosIdentity.psm1 (hunk 1 above)
Comment (hckenmiller) [reply Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: It is trimming inside of the Reset-ApiPassword

File: Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1 (hunk 2 above)
Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1]: good catch

File: Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1 [hunk 3]
```diff
@@ -106,15 +105,151 @@ Describe 'Identity Cli Functional Tests' {
...
+                $jsonApi = $newApiResource | ConvertTo-Json
+
+                $editApi = Edit-ApiRegistration -identityUrl $identityUrl -body $jsonApi -apiName "test-Api" -accessToken $accessToken
+
+                $editApi | Should -Not -Be $null
+
+                # Cleanup the registered Api
+                $removeApi = Remove-ApiRegistration -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $removeApi | Should -Be ""
```
Comment (michaelvidal24) [inline Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1]: To really make this test effective, should we try a get on the removed api and ensure we get a 404?

Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1]: I have another test in this group that does a Get, Edit and Remove on a non-existent api that return not found.

File: Fabric.Identity.API/scripts/CatalystDosIdentity.psm1 [hunk 4]
```diff
@@ -599,6 +599,10 @@ function New-ApiRegistration {
     }
 
     try{
+        if(!($body -is [string])) {
+            $body = $body | ConvertTo-Json
```
Comment (ryanorbaker) [inline Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: is this passed by reference?  if so we are changing the value of on input parameter.  Should we be using a local variable?

Comment (ryanorbaker) [reply Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: nevermind, powershell passes by value.

**Pull request merged**